apps/web/__e2e__/sorting-notebooks.test.ts

```diff
@@ -0,0 +1,205 @@
+    groupBy: "none",
+    orderBy: "ascendingOrder",
+    sortBy: "dateCreated"
+  });
+  expect(await isLengthCorrect(page)).toBeTruthy();
+  await notebooks.sort({
+    groupBy: "none",
+    orderBy: "descendingOrder",
+    sortBy: "dateCreated"
+  });
+  expect(await isLengthCorrect(page)).toBeTruthy();
+  await notebooks.sort({
+    groupBy: "week",
+    orderBy: "ascendingOrder",
+    sortBy: "dateCreated"
+  });
+  expect(await isLengthCorrect(page)).toBeTruthy();
+  await notebooks.sort({
+    groupBy: "week",
+    orderBy: "descendingOrder",
+    sortBy: "dateCreated"
+  });
+  expect(await isLengthCorrect(page)).toBeTruthy();
+  await notebooks.sort({
+    groupBy: "year",
+    orderBy: "ascendingOrder",
+    sortBy: "dateCreated"
+  });
+  expect(await isLengthCorrect(page)).toBeTruthy();
+  await notebooks.sort({
+    groupBy: "year",
+    orderBy: "descendingOrder",
+    sortBy: "dateCreated"
+  });
+  expect(await isLengthCorrect(page)).toBeTruthy();
+
+  //dateEdited
+  await notebooks.sort({
+    groupBy: "abc",
+    orderBy: "ascendingOrder",
+    sortBy: "dateEdited"
+  });
+  expect(await isLengthCorrect(page)).toBeTruthy();
+  await notebooks.sort({
+    groupBy: "abc",
+    orderBy: "descendingOrder",
+    sortBy: "dateEdited"
+  });
+  expect(await isLengthCorrect(page)).toBeTruthy();
+  await notebooks.sort({
+    groupBy: "default",
+    orderBy: "ascendingOrder",
+    sortBy: "dateEdited"
+  });
+  expect(await isLengthCorrect(page)).toBeTruthy();
+  await notebooks.sort({
+    groupBy: "default",
+    orderBy: "descendingOrder",
+    sortBy: "dateEdited"
+  });
+  expect(await isLengthCorrect(page)).toBeTruthy();
+  await notebooks.sort({
+    groupBy: "month",
+    orderBy: "ascendingOrder",
+    sortBy: "dateEdited"
+  });
+  expect(await isLengthCorrect(page)).toBeTruthy();
+  await notebooks.sort({
+    groupBy: "month",
+    orderBy: "descendingOrder",
+    sortBy: "dateEdited"
+  });
+  expect(await isLengthCorrect(page)).toBeTruthy();
+  await notebooks.sort({
+    groupBy: "none",
+    orderBy: "ascendingOrder",
+    sortBy: "dateEdited"
+  });
+  expect(await isLengthCorrect(page)).toBeTruthy();
+  await notebooks.sort({
+    groupBy: "none",
+    orderBy: "descendingOrder",
+    sortBy: "dateEdited"
+  });
+  expect(await isLengthCorrect(page)).toBeTruthy();
+  await notebooks.sort({
+    groupBy: "week",
+    orderBy: "ascendingOrder",
+    sortBy: "dateEdited"
+  });
+  expect(await isLengthCorrect(page)).toBeTruthy();
+  await notebooks.sort({
+    groupBy: "week",
+    orderBy: "descendingOrder",
+    sortBy: "dateEdited"
+  });
+  expect(await isLengthCorrect(page)).toBeTruthy();
+  await notebooks.sort({
+    groupBy: "year",
+    orderBy: "ascendingOrder",
+    sortBy: "dateEdited"
+  });
+  expect(await isLengthCorrect(page)).toBeTruthy();
+  await notebooks.sort({
+    groupBy: "year",
+    orderBy: "descendingOrder",
+    sortBy: "dateEdited"
+  });
+  expect(await isLengthCorrect(page)).toBeTruthy();
```

Comment:
Ideally tests should follow the AAA rule (Arrange, Act, Assert). Here's what you should do:

1. Move all possible Group By, Sort by & Order by options into one array each.
2. Use 3 nested loops to iterate over all possible sorting combinations.
3. Inside the nested loops should be 1 test that will follow the AAA pattern (create notes, do the sorting, ensure items are there).